data_structures/src/chain.rs

```diff
@@ -639,64 +706,180 @@ impl TransactionsPool {
+    /// pool.insert(transaction.clone());
     ///
-    /// assert!(pool.contains(&hash));
+    /// assert!(pool.contains(&transaction.hash()));
     ///
-    /// let op_transaction_removed = pool.remove(&hash);
+    /// let op_transaction_removed = pool.vt_remove(&transaction.hash());
     ///
-    /// assert_eq!(Some(transaction), op_transaction_removed);
-    /// assert!(!pool.contains(&hash));
+    /// assert_eq!(Some(vt_transaction), op_transaction_removed);
+    /// assert!(!pool.contains(&transaction.hash()));
     /// ```
-    pub fn remove(&mut self, key: &Hash) -> Option<Transaction> {
-        self.transactions.remove(key).map(|(weight, transaction)| {
-            self.sorted_index.remove(&(weight, *key));
-            transaction
-        })
+    pub fn vt_remove(&mut self, key: &Hash) -> Option<VTTransaction> {
+        self.vt_transactions
+            .remove(key)
+            .map(|(weight, transaction)| {
+                self.sorted_index.remove(&(weight, *key));
+                transaction
+            })
+    }
+
+    /// Returns an `Option` with the data request transaction for the specified hash or `None` if not exist.
+    ///
+    /// The `key` may be any borrowed form of the hash, but `Hash` and
+    /// `Eq` on the borrowed form must match those for the key type.
+    ///
+    /// # Examples:
+    /// ```
+    /// # use witnet_data_structures::chain::{TransactionsPool, Hash, Hashable};
+    /// # use witnet_data_structures::transaction::{Transaction, DRTransaction};
+    /// let mut pool = TransactionsPool::new();
+    /// let dr_transaction = DRTransaction::default();
+    /// let transaction = Transaction::DataRequest(dr_transaction.clone());
+    /// pool.insert(transaction.clone());
+    ///
+    /// assert!(pool.contains(&transaction.hash()));
+    ///
+    /// let op_transaction_removed = pool.dr_remove(&transaction.hash());
+    ///
+    /// assert_eq!(Some(dr_transaction), op_transaction_removed);
+    /// assert!(!pool.contains(&transaction.hash()));
+    /// ```
+    pub fn dr_remove(&mut self, key: &Hash) -> Option<DRTransaction> {
+        self.dr_transactions.remove(key)
+    }
+
+    /// Returns a tuple with a vector of commit transactions that achieve the minimum specify
+    /// by the data request, and the value of all the fees obtained with those commits
+    pub fn remove_commits(&mut self, dr_pool: &DataRequestPool) -> (Vec<CommitTransaction>, u64) {
+        let mut total_fee = 0;
+        let commits_vector = self
+            .co_transactions
+            .iter_mut()
+            .map(|(dr_pointer, commits)| {
+                let mut commits_vec = vec![];
+
+                if let Some(dr_output) = dr_pool.get_dr_output(&dr_pointer) {
+                    let n_commits = dr_output.witnesses as usize;
+
+                    if commits.len() >= n_commits {
+                        for (i, (_h, c)) in commits.drain().enumerate() {
```

Comment:
```suggestion
                        for (_h, c) in commits.drain().take(n_commits) {
```

Reply:
Or even better, something like:

```suggestion
let commits_vec = commits.drain().take(n_commits)).collect();
```